proof/refine/X64/ArchCSpace1_R.thy

```diff
@@ -1140,6 +1152,25 @@ lemma valid_badges_preserve_oneway[CSpace1_R_2_assms]:
   apply (clarsimp simp: valid_arch_badges_def)
   done
 
+(* corresponds to safe_parent_for_arch in AInvs, used by safe_parent_for' *)
+definition
+  "safe_parent_for_arch' m cap parent \<equiv>
+     isArchIOPortCap cap \<and> isIOPortControlCap' parent \<and> (\<forall>p n'. m p \<noteq> Some (CTE cap n'))"
```

Comment:
That is really working out quite nicely. Cool.

Reply:
Inspired by what happened in AInvs, but had to have different definitions, had to add the `m` parameter, and couldn't add a sane type signature because I didn't know what the bits were.

Reply:
That's ok for now, we can a type signature later if we want it. I'm happy that we managed to split this one at all.